vcf.c

```diff
@@ -645,6 +650,35 @@ int bcf_hdr_parse(bcf_hdr_t *hdr, char *htxt)
         needs_sync += bcf_hdr_add_hrec(hdr, hrec);
         p += len;
     }
+
+    // Detect incorrect header lines, print a warning, and skip them.
+    // Alternatively, the program could refuse to work with such VCF. However,
+    // many VCF operations do not really care about a few malformed header
+    // lines.
+    if ( strncmp("#CHROM\tPOS",p,10) )
```

Comment:
... Or maybe not.  bcftools calls `bcf_hdr_add_sample()` in loads of places without checking the return value.  I'll leave this one for later.